bin/fin

```diff
@@ -5083,28 +5084,98 @@ addon () {
 
 	case "$1" in
 		install|in)
-				addon_install "$addon_name" ${is_global}
-			;;
+			addon_install "$addon_name" ${is_global}
+		;;
 		remove|rm)
 			addon_remove "$addon_name" ${is_global}
-			;;
+		;;
+		update|up)
+			addon_update "$addon_name" ${is_global}
+		;;
 		version|v)
-			# special case for addon version command
+		# special case for addon version command
 			addon_script=$(get_addon_script "$addon_name")
 			[ ! -f "$addon_script" ] && echo "-1" && exit
 			addon_version=$(cat "$addon_script" | grep ^VERSION | cut -d = -f 2 | tr -d \" | tr -d \')
 			addon_version="${addon_version:-0}"
 			echo "$addon_version"
-			;;
+		;;
 		help)
 			show_help_addon
-			;;
+		;;
 		*)
 			echo-yellow "Unknown sub-command. See ${NC}fin addon help${yellow} for details."
-			;;
+		;;
 	esac
 }
 
+# Check to see if addon exists within remote repo
+# $1 - uuid to use for foldername in /tmp
+# $2 - addon_name the folder/file to find
+# $3 - repo_url the repository url to check
+# $4 - repo_branch the branch to check against
+_addon_temp_install () {
+	local uuid=${1}
+	local addon_name=${2}
+	local repo_url=${3}
+	local repo_branch=${4}
+	local tmp_random="/tmp/${uuid}"
+	mkdir $tmp_random && cd $tmp_random
+
+	# Initialize empty repo; suppress output
+	git init > /dev/null
+	git remote add origin $repo_url > /dev/null
+	git fetch -q origin $repo_branch > /dev/null
+	git cat-file -s origin/${repo_branch}:${addon_name} 2> /dev/null
+	git checkout origin/${repo_branch} ${addon_name} 2> /dev/null
+}
+
+# Search all of the provided repositories for the requested plugin.
+_addon_search_repo () {
+	eval 'repos=(${!'"DOCKSAL_ADDONS_REPO"'@})'
+
+	local addon=$1
+	local repo_url=$2
+	local repo_branch=$3
+	local repo_uuid=$4
+
+	echo-green "Searching for ${addon} in predefined repositories..."
+	for key in "${repos[@]}"
+	do
+		local config=${!key}
+		# Append master just incase there is no branch defined
+		config="${config}|master"
+		local REPO_URL=$(echo $config | cut -d'|' -f1)
+		local REPO_BRANCH=$(echo $config | cut -d'|' -f2)
+		local uuid=$(uuidgen)
+
+		# Check to make sure the provided Repo URL is valid and the Branch Exists
+		git ls-remote -h "$REPO_URL" --refs "$REPO_BRANCH" 2>/dev/null | grep -q "$REPO_BRANCH"
+		if [[ "$?" != "0" ]]; then
+			echo "$key defines an invalid repository/branch reference. Skipping..."
+			continue
+		else
+			local exists=$(_addon_temp_install $uuid $addon $REPO_URL $REPO_BRANCH)
+			if [[ -n "${exists}" ]] && [[ $exists -ge 0 ]]; then
+				if _confirm "Addon Found: Install ${NC}${key}/${addon}" "--no-exit" -eq 0; then
+					eval "${repo_url}=${REPO_URL}"
+					eval "${repo_branch}=${REPO_BRANCH}"
+					eval "${repo_uuid}=${uuid}"
+					break;
```

Comment:
`return` ?

Reply:
@achekulaev I'm not removing the folder here with the though that if they have chosen to install the module we will have the data downloaded and no need to re-clone it. Which is why I am passing the UUID as a reference so we know the folder to use.